File: swedish-cc/2025/lovable-challenge/encryption/encryption.py
s = input()
d = input()
lcs = input()

i = j = k = 0
out = []

while i < len(s) or j < len(d):
    if i >= len(s):
        out += [d[j]]
        j += 1
        continue

    if j >= len(d):
        out += [s[i]]
        i += 1
        continue

    if k >= len(lcs):
        out += [s[i]]
        i += 1
        continue

    if lcs[k] == s[i] == d[j]:
        out += [lcs[k]]
        k += 1
        i += 1
        j += 1
        continue

    if lcs[k] == s[i]:
        out += [d[j]]
        j += 1
        continue

    out += [s[i]]
    i += 1

print("".join(out))

File: swedish-cc/2025/lovable-challenge/encryption/encryption.cpp
#include <bits/stdc++.h>
using namespace std;

#define sz(c) ((int)c.size())

int main() {
  cin.tie(0)->sync_with_stdio(0); 

  string s, d, lcs;
  cin >> s >> d >> lcs;

  int i = 0;
  int j = 0;
  int k = 0;

  string res;

  while (i < sz(s) || j < sz(d)) {
    if (i >= sz(s)) {
      res += d[j];
      j++;
      continue;
    }

    if (j >= sz(d)) {
      res += s[i];
      i++;
      continue;
    }

    if (k >= sz(lcs)) {
      res += s[i];
      i++;
      continue;
    }

    if (lcs[k] == s[i] && lcs[k] == d[j]) {
      res += lcs[k];
      k++;
      i++;
      j++;
      continue;
    }

    if (lcs[k] == s[i]) {
      res += d[j];
      j++;
      continue;
    }

    res += s[i];
    i++;
  }

  cout << res;
}
